system-addon/test/unit/lib/Store.test.js

```diff
@@ -0,0 +1,83 @@
+const {Store} = require("lib/Store.jsm")
+const {actionTypes: at} = require("common/Actions.jsm");
+const {createStore} = require("redux");
+const {addNumberReducer} = require("test/unit/utils");
+
+describe("Store", () => {
+  let sandbox;
+  let store;
+  before(() => {
+    sandbox = sinon.sandbox.create();
+  });
+  beforeEach(() => {
+    store = new Store();
+  });
+  afterEach(() => sandbox.restore());
+  it("should have an empty feeds Set", () => {
+    assert.instanceOf(store.feeds, Set);
+    assert.equal(store.feeds.size, 0, ".feeds.size");
+  });
+  it("should have a redux store at ._store", () => {
+    assert.ok(store._store);
+  });
+  describe("#init", () => {
+    it("should dispatch an init action", () => {
+      sandbox.spy(store, "dispatch");
+      store.init();
+      assert.calledWith(store.dispatch, {type: at.INIT});
+    });
+    it("should dispatch an init action", () => {
+      sandbox.spy(store, "dispatch");
+      store.init([]);
+      assert.calledWith(store.dispatch, {type: at.INIT});
+    });
+    it("should add each subscriber to .actionScribers and .store to each subscriber", () => {
+      const s = {};
+      store.init([s]);
+      assert.isTrue(store.feeds.has(s), ".feeds.has(s)");
+      assert.equal(s.store, store, "s.store");
+    });
+  });
+  describe("#uninit", () => {
+    it("should clear .feeds", () => {
+      store.init([{}, {}, {}]);
+      assert.equal(store.feeds.size, 3);
+      store.uninit();
+      assert.equal(store.feeds.size, 0);
+    });
+    it("should dispatch an uninit action", () => {
+      sandbox.spy(store, "dispatch");
+      store.uninit();
+      assert.calledWith(store.dispatch, {type: at.UNINIT});
+    });
+  });
+  describe("#getState", () => {
+    it("should return the redux state", () => {
+      store._store = createStore((prevState = 123) => prevState);
+      assert.equal(store.getState(), 123);
+    });
+  });
+  describe("#dispatch", () => {
+    it("should call .onAction of each feed", () => {
+      const sub = {onAction: sinon.spy()};
+      const action = {type: "FOO"};
+      store.init([sub]);
+      store.dispatch(action);
+      assert.calledWith(sub.onAction, action);
+    });
```

Comment:
One convention that I've found makes non-trivial tests substantially easier to skim is to split them up with whitespace into three sections: initialize, execute, verify.  In this case, that would look like:

```js
      const sub = {onAction: sinon.spy()};
      const action = {type: "FOO"};
      store.init([sub]);

      store.dispatch(action);

      assert.calledWith(sub.onAction, action);
```